keps/draft-repo.md

```diff
@@ -0,0 +1,108 @@
+
+By providing a repository (or set of repositories), KUDO will provide a mechanism for administrators to easy discover and install KUDO operators.  [Helm has provided some prior art](https://github.com/helm/helm/blob/master/docs/chart_repository.md) which we should align closely to from a mental model stand point.
+It is useful to note that helm is [moving away from the custom repository model](https://blog.bacongobbler.com/post/2019-01-25-distributing-with-distribution/index.html) and towards OCI-compatible registries for a backend.  While this seems like a worthy ambition, it adds work that we should focus on for a phase 2 effort.   The first version of the repository should
+
+## Motivation
+
+We need a way for the operator author to update a repository with their operator.  We also need a way for the operator to promote an operator as being production ready. Finally we need a KUDO user to point to different repositories.  
+
+### Goals
+
+- Automate the creation of a operator tarball
+- Automate the creation or updating of the repository index with a new operator tarball
+- Define rules for incubator and graduated along with graduation tooling.
+- Define how an admin can create their own repository
+- Install operator from incubator instead of graduated repository
+
+### Non-Goals
+
+- Manage repos of non-KUDO applications (e.g. Helm Charts)
+- External formats defined by [KEP0013](keps/0013-external-specs.md)
+- Repository cacheing
+
+## Proposal
+
+### Create Operator Tarball
+
+An operator packaage as defined in [KEP0010](keps/0010-package-manager.md).  The storage package at the repository is a tarball. We need a way to create this tarball in a uniformed way. KUDO shall a way to generate a tarball based on the standard file system layout.  As an example: `kubectl kudo package docs/examples/zookeeper/`.  This will create a `zookeeper-3.4.10.tgz` based on parsing the `operator.yaml` for version details. It would be best if this included linting to ensure that the static structure of the operator is correct. At this point, it will guarantee that operator folder has `operator.yaml` and `params.yaml` along with a templates directory.  It will create the following operator tarball `{operator.yaml:name}-{operator.yaml:version}.tgz`.
+
+### Create Repository Index
+
+KUDO needs the ability to create an index file for the repository. Something like `kubectl kudo repo index new-repo --url https://kudo-repo.storage.googleapis.com`. In this example, the `kudo repo index` is the command. The `new-repo` is a folder containing operator tarballs. The index file defined in [KEP0010](keps/0010-package-manager.md) will be created using the url provided for links in the file.
+
+Steps for creating an index file for a new operator looks like:
+```
+mkdir new-repo
+kubectl kudo package docs/examples/zookeeper/ --destination=new-rep
+kubectl kudo repo index new-repo --url https://kudo-repo.storage.googleapis.com
+```
+
+### Create Local Respository
```

Comment:
this is for me out of scope for this KEP, I would remove it. It does not even address any of the goals. I think we need to put a more thought into running local/custom repositories, I would almost see that as a separate KEP

Reply:
I would love our keps to be very lean and something we commit to in one release iteration and I don't see this as an essential feature, just nice to have. Does not really help with Day 2 or our core business so I would postpone it and remove it but if you feel like it's very high on our roadmap 🤷‍♀ just keep in mind every feature mean implementing it, testing it and documenting it